core/src/main/java/brooklyn/util/crypto/SecureKeys.java

```diff
@@ -19,182 +19,149 @@
+    /** RFC1773 order, with None for other values. Normally prefer X500Principal. */
+    public static X509Principal getX509PrincipalWithCommonName(String commonName) {
+        return new X509Principal("" + "C=None," + "L=None," + "O=None," + "OU=None," + "CN=" + commonName);
     }
 
-    public static KeyPair newKeyPair(String algorithm, int bits) {
-        return newKeyPairGenerator(algorithm, bits).generateKeyPair();
-    }
+    /** reads RSA or DSA / pem style private key files (viz {@link #toPem(KeyPair)}), extracting also the public key if possible
+     * @throws IllegalStateException on errors, in particular {@link PassphraseProblem} if that is the problem */
+    public static KeyPair readPem(InputStream input, final String passphrase) {
+        // TODO cache is only for fallback "reader" strategy 
+        byte[] cache = Streams.readFully(input);
+        input = new ByteArrayInputStream(cache);
 
-    /** returns a new keystore, of the default type, and initialized to be empty.
-     * (everyone always forgets to load(null,null).) */
-    public static KeyStore newKeyStore() {
         try {
-            KeyStore store = KeyStore.getInstance(KeyStore.getDefaultType());
-            store.load(null, null);
-            return store;
-        } catch (Exception e) {
-            throw Exceptions.propagate(e);
-        }
-    }
+            PEMParser pemParser = new PEMParser(new InputStreamReader(input));
+
+            Object object = pemParser.readObject();
+            pemParser.close();
+
+            JcaPEMKeyConverter converter = new JcaPEMKeyConverter().setProvider("BC");
+            KeyPair kp = null;
+            if (object==null) {
+                throw new IllegalStateException("PEM parsing failed: missing or invalid data");
+            } else if (object instanceof PEMEncryptedKeyPair) {
+                if (passphrase==null) throw new PassphraseProblem("passphrase required");
+                try {
+                    PEMDecryptorProvider decProv = new JcePEMDecryptorProviderBuilder().build(passphrase.toCharArray());
+                    kp = converter.getKeyPair(((PEMEncryptedKeyPair) object).decryptKeyPair(decProv));
+                } catch (Exception e) {
+                    Exceptions.propagateIfFatal(e);
+                    throw new PassphraseProblem("wrong passphrase");
+                }
+            } else  if (object instanceof PEMKeyPair) {
+                kp = converter.getKeyPair((PEMKeyPair) object);
+            } else if (object instanceof PrivateKeyInfo) {
+                PrivateKey privKey = converter.getPrivateKey((PrivateKeyInfo) object);
+                kp = new KeyPair(null, privKey);
+            } else {
+                throw new IllegalStateException("PEM parser support missing for: "+object);
+            }
+
+            return kp;
 
-    /** returns keystore of default type read from given source */
-    public static KeyStore newKeyStore(InputStream source, String passphrase) {
```

Comment:
Unfortunately these methods (e.g. `newKeyStore`, `getTrustManager`, `getCanonicalImpl` etc) weren't marked as `@Beta` and neither was the class. Are we definitely ok with just deleting them rather than deprecating them?